pkg/pub_integration/lib/src/fake_test_user.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:_pub_shared/pubapi.dart';
+import 'package:http/http.dart' as http;
+import 'package:pub_integration/src/fake_credentials.dart';
+import 'package:pub_integration/src/headless_env.dart';
+import 'package:pub_integration/src/pub_puppeteer_helpers.dart';
+import 'package:puppeteer/puppeteer.dart';
+
+import 'test_scenario.dart';
+
+class FakeTestUser implements TestUser {
+  @override
+  final String email;
+
+  @override
+  final Browser browser;
+
+  @override
+  final PubApiClient api;
+
+  FakeTestUser({
+    required this.email,
+    required this.browser,
+    required this.api,
+  });
+
+  @override
+  Future<String> readLatestEmail() {
+    // TODO: implement readLatestEmail after #6549 lands.
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<Map<String, Object?>> createCredentials() async {
+    return fakeCredentialsMap(email: email);
+  }
+
+  static Future<FakeTestUser> create({
```

Comment:
I'd suggest making this a top-level `createFakeUser` function instead, so we don't have to make `FakeTestUser` a public class.